resource-aggregate/events/resourceLinksSnapshotTaken.go

```diff
@@ -146,9 +166,9 @@ func (e *ResourceLinksSnapshotTaken) HandleCommand(ctx context.Context, cmd aggr
 			AuditContext:  ac,
 			EventMetadata: em,
 		}
-		err := e.HandleEventResourceLinksPublished(ctx, &rlp)
-		if err != nil {
-			return nil, err
+		ok := e.HandleEventResourceLinksPublished(ctx, &rlp)
```

Comment:
&rlp is sent to the eventStore and eventBus. As far as I understand your code, it will every time send and event which contains also resourceLinks which were already published. Shouldn't the final event contain only resource links which were published within one command execution?

That means, `HandleEventResourceLinksPublished` not only verifies and appends the dif to snapshot, but also returns resources which were published. Currently it returns true so you will send an event which contains resources which were already published. 

In case of unpublish, unpublished events are deleted from the snapshot struct and items which were deleted are returned to the eventstore/eventbus.
In my opinion same should happen for the publish, what was added is appended (this is done), and only published resources are returned to the eventstore/eventbus.
Please confirm @jkralik

Reply:
You are correct, I didn't commit all the changes I did by mistake.